test/refactoring/ConvertAsync/convert_async_renames.swift

```diff
@@ -0,0 +1,353 @@
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+2):3 | %FileCheck -check-prefix=SHADOWINNERBIND %s
+func shadowInnerBind() {
+  simple { str in
+    simpleErr(arg: str) { other, err in
+      if let str = other {
+        print(str)
+      }
+    }
+  }
+}
+// SHADOWINNERBIND: let str = await simple()
+// SHADOWINNERBIND-NEXT: let str0 = try await simpleErr(arg: str)
+// SHADOWINNERBIND-NEXT: print(str0)
+
+// TODO: `throws` isn't added to the function declaration
+// RUN: %refactor  -convert-to-async -dump-text -source-filename %s -pos=%(line+2):6 | %FileCheck -check-prefix=SHADOWNAMEEXISTS %s
+// RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+2):3 | %FileCheck -check-prefix=SHADOWNAMEEXISTS %s
+func shadowNameAlreadyExists() {
+  simple { str in
+    simpleErr(arg: str) { str, err in
+      let str0 = "str0"
+      print(str0)
+      if let str0 = str {
+        print(str0)
+      }
+      if let str1 = str {
+        print(str1)
+      }
+    }
+  }
+}
+// SHADOWNAMEEXISTS: let str = await simple()
+// SHADOWNAMEEXISTS-NEXT: let str0 = try await simpleErr(arg: str)
+// SHADOWNAMEEXISTS-NEXT: let str00 = "str0"
+// SHADOWNAMEEXISTS-NEXT: print(str00)
+// SHADOWNAMEEXISTS-NEXT: print(str0)
+// SHADOWNAMEEXISTS-NEXT: print(str0)
+
+func shadowsUsedDecl() {
+  let inOuter: String = "str"
+  print(inOuter)
+  // TODO: `throws` isn't added to the function declaration
+  // RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+2):8 | %FileCheck -check-prefix=SHADOWOUTERUSED %s
+  // RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+2):10 | %FileCheck -check-prefix=SHADOWOUTERUSED %s
+  func inner() {
+    simpleErr(arg: "str") { inCall, err in
+      if inCall != nil {
+        let inOuter = inCall!
+        print(inOuter)
+      }
+      print(inOuter)
+    }
+  }
+}
+// SHADOWOUTERUSED: let inCall = try await simpleErr(arg: "str")
+// SHADOWOUTERUSED-NEXT: let inOuter0 = inCall
+// SHADOWOUTERUSED-NEXT: print(inOuter0)
+// SHADOWOUTERUSED-NEXT: print(inOuter)
+
+func shadowsUnusedDecl() {
+  let inOuter: String = "str"
+  print(inOuter)
+  // TODO: `throws` isn't added to the function declaration
+  // RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+2):8 | %FileCheck -check-prefix=SHADOWOUTERUNUSED %s
+  // RUN: %refactor -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+2):10 | %FileCheck -check-prefix=SHADOWOUTERUNUSED %s
+  func inner() {
+    simpleErr(arg: "str") { inCall, err in
+      if inCall != nil {
+        let inOuter = inCall!
+        print(inOuter)
+      }
+    }
+  }
+}
+// SHADOWOUTERUNUSED: let inCall = try await simpleErr(arg: "str")
+// SHADOWOUTERUNUSED-NEXT: let inOuter = inCall
+// SHADOWOUTERUNUSED-NEXT: print(inOuter)
```

Comment:
Would it be worth adding a test for a top-level name conflict?

Reply:
Yeah, probably - will do, thanks.

Reply:
Good catch! It actually failed since top level code is wrapped in a TopLevelCodeDecl and implicit BraceStmt, which caused the DeclCollector to miss those decls.